### PR TITLE
Np 3497 authenticated cristin client

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/client/ApiClient.java [hunk 1]
```diff
@@ -76,6 +73,24 @@ public ApiClient(HttpClient client) {
         return getSuccessfulResponseOrThrowException(httpRequest);
     }
 
+    public HttpResponse<String> fetchPostResult(URI uri, String body) {
+        HttpRequest httpRequest = HttpRequest.newBuilder()
+                .uri(uri)
+                .header("Content-Type", "application/json")
```
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/cristin/common/client/ApiClient.java:79]: Use standard constants

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/client/ApiClient.java [hunk 2]
```diff
@@ -76,6 +73,24 @@ public ApiClient(HttpClient client) {
...
+                .POST(HttpRequest.BodyPublishers.ofString(body))
+                .build();
+        return attempt(() -> client.send(httpRequest, BodyHandlers.ofString(StandardCharsets.UTF_8))).orElseThrow();
+    }
+
+    public CompletableFuture<HttpResponse<String>> fetchPostResultAsync(URI uri, String body) {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(uri)
+                .header("Content-Type", "application/json")
```
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/cristin/common/client/ApiClient.java:88]: Ditto

File: cristin-project/src/test/java/no/unit/nva/cristin/common/client/AuthenticatedApiClientTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+class AuthenticatedApiClientTest {
+
+    private ApiClient cristinApiClient;
+
+    @BeforeEach
+    void setUp() {
+        cristinApiClient = new ApiClient(CristinAuthenticator.getHttpClient());
+    }
+
+    //    @Test
```
Comment (brinxmat) [inline cristin-project/src/test/java/no/unit/nva/cristin/common/client/AuthenticatedApiClientTest.java:25]: Since we're not testing, is the test needed?

**Review by brinxmat — APPROVED** [review]

**Review by brinxmat — CHANGES_REQUESTED** [review]
See commentss

File: cristin-project/src/test/java/no/unit/nva/cristin/common/client/AuthenticatedApiClientTest.java (hunk 3 above)
Comment (svenngjeto) [reply cristin-project/src/test/java/no/unit/nva/cristin/common/client/AuthenticatedApiClientTest.java]: Removed test, added simple javadoc instead

**Review by brinxmat — APPROVED** [review]

**Pull request merged**